core/src/main/java/me/nov/threadtear/decompiler/DecompilerInfo.java

```diff
@@ -0,0 +1,26 @@
+package me.nov.threadtear.decompiler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class DecompilerInfo<T extends IDecompilerBridge> {
+
+  public abstract String getName();
+
+  public abstract String getVersionInfo();
+
+  public abstract T createDecompilerBridge();
+
+  @Override
+  public String toString() {
+    return getName() + " " + getVersionInfo();
+  }
+
+  public static List<DecompilerInfo<?>> getDecompilerInfos() {
```

Comment:
This isn't still ideal as they are still hard coded. Ideally this would be done by creating a provider class that loads all available `DecompilerInfo` classes using the `ServiceLoader`.